test/unit/test_utils.py

```diff
@@ -0,0 +1,104 @@
+import os
+import shutil
+from pathlib import Path
+
+import pytest
+
+from ansible_builder.utils import write_file, copy_file, run_command
+
+
+def test_write_file(tmpdir):
+    path = os.path.join(tmpdir, 'foo.txt')
+    text = [
+        'foo  # from some collection',
+        'bar',
+        '# a comment',
+        '',
+        'zoo',
+        ''  # trailing line
+    ]
+    assert write_file(path, text)  # does not exist, write
+    assert not write_file(path, text)  # already correct, do not write
+
+
+@pytest.fixture
+def source_file(tmpdir):
+    source = os.path.join(tmpdir, 'bar.txt')
+    with open(source, 'w') as f:
+        f.write('foo\nbar\n')
+    return source
```

Comment:
Why use tmp_dir?

```suggestion
def source_file(tmp_path):
    source = tmp_path / 'bar.txt'
    source.write_text('foo\nbar\n')
    return str(source)
```

Reply:
I'm not sure I fully understand the question. Why use a temporary directory at all here? Are you saying this should just be a permanent test fixture rather than creating the same file each time? Or should I return a `Path` object if I'm going to bother changing this to `tmp_path`?

I didn't look at the broader use of the fixture and how it's used it tests, so there is most likely more room for improvement.

Reply:
This is mostly about the `tmpdir` fixture being deprecated while `tmp_path` is the thing that replaces it and should be always used instead.

Reply:
Oh man, I had a total Monday moment. Sorry. I didn't realize that was a code suggestion. I thought it was a diff from my PR. 🤦‍♂️ 

I have switched to using `tmp_path` in _so many_ places I thought that was a change I had made here as well.

Reply:
These fixtures are all original code I just moved. That's why they are showing up as "new" changes in this PR.

Ok, now I think I've un-confused myself. 😄